script.js

```diff
@@ -0,0 +1,75 @@
+const title = document.getElementById('title');
+const author = document.getElementById('author');
+const addBtn = document.getElementById('add');
+const div = document.querySelector('.book');
+
+let bookArr = [];
+
+function addBooks() {
+  const myBooks = {
+    title: title.value,
+    author: author.value,
+    id: bookArr.length + 1,
+  };
+  if (title.value === '' || author.value === '') { return; }
+
+  bookArr.push(myBooks);
+}
+
+function storeData() {
+  localStorage.setItem('bookCollection', JSON.stringify(bookArr));
+}
+
+function getData() {
+  bookArr = JSON.parse(localStorage.getItem('bookCollection'));
+}
```

Comment:
- [ ] Kindly try to fix that error 
![image](https://user-images.githubusercontent.com/24703171/176206288-ecd22dd3-6b2d-459a-9ea4-bf93ff165a82.png)
this problem happens if the key `bookCollection` doesn't exist in local storage,
you can fix it by changing this line to be like this
```
bookArr = JSON.parse(localStorage.getItem('bookCollection')) || [ ] ;
```

Reply:
Okay, will change that. Thanks for the info. Learned again something. 😄